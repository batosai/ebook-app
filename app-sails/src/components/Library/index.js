import React, { Component, PropTypes as T } from 'react';
import { GridList, GridTile } from 'material-ui/GridList';
import { Link } from 'react-router';
import { config } from '../../config';

class Library extends Component {
  render = () => {
    return (
      <div style={this.props.style.root}>
        <GridList cellHeight={180} style={this.props.style.gridList}>
          {this.props.tiles.map(tile => (
            <Link key={tile.id} to={`/collection/${tile.id}`}>
              <GridTile
                title={tile.title}
                subtitle={
                  <span>
                    by <b>{tile.author}</b>
                  </span>
                }
              >
                <img src={`${config.server}/collections/${tile.id}/illustration`} alt="" />
              </GridTile>
            </Link>
          ))}
        </GridList>
      </div>
    );
  };
}

Library.propTypes = {
  style: T.object.isRequired,
  tiles: T.arrayOf(T.object).isRequired,
};

export default Library;
