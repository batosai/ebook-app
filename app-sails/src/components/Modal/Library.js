import React, { Component, PropTypes as T } from 'react';
import TextField from 'material-ui/TextField';
import Dialog from 'material-ui/Dialog';

import modalActions from './modalActions';

class ModalLibrary extends Component {
  state = {
    id: null,
    name: '',
  };

  componentDidUpdate = (prevProps, prevState) => {
    if (prevProps.library !== this.props.library) {
      this.setState({
        id: this.props.library.id,
        name: this.props.library.name,
      });
    }
  };

  handleClose = () => {
    this.setState({
      id: null,
      name: '',
    });
    this.props.onRequestClose();
  };

  handleChange = event => {
    this.setState({
      name: event.target.value,
    });
  };

  save = () => {
    if (this.state.id) {
      this.props.editLibrary(this.state);
    } else {
      this.props.createLibrary(this.state);
    }

    this.props.onRequestClose();
    this.setState({
      id: null,
      name: '',
    });
  };

  render = () => {
    const actions = modalActions(this.handleClose, this.save);

    return (
      <Dialog
        title={this.state.id ? 'Edit library' : 'Add library'}
        actions={actions}
        modal={false}
        open={this.props.open}
        onRequestClose={this.handleClose}
      >
        <TextField
          autoFocus
          hintText="Name"
          floatingLabelText="Name"
          errorText="This name is required"
          fullWidth={true}
          value={this.state.name}
          onChange={this.handleChange}
        />
      </Dialog>
    );
  };
}

ModalLibrary.propTypes = {
  open: T.bool.isRequired,
  createLibrary: T.func.isRequired,
  editLibrary: T.func.isRequired,
  library: T.object,
};

export default ModalLibrary;
