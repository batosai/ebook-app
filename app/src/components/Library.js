import React, { Component, PropTypes as T } from 'react';
import { GridList, GridTile } from 'material-ui/GridList';
import { Link } from 'react-router';

class Library extends Component {

  render() {
    return (
      <div style={this.props.style.root}>
        <GridList
          cellHeight={180}
          style={this.props.style.gridList} >
          {this.props.tiles.map((tile) => (
            <Link key={tile.id} to={`/collection/${tile.id}`}><GridTile
              title={tile.title}
              subtitle={<span>by <b>{tile.author}</b></span>} >
              <img src={tile.img} alt="" />
            </GridTile></Link>
          ))}
        </GridList>
      </div>
    );
  }
}

Library.propTypes = {
  style: T.object.isRequired,
  tiles: T.arrayOf(T.object).isRequired
};

export default Library;
