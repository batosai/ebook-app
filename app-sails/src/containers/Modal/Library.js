import { connect } from 'react-redux';
import Library from '../../components/Modal/Library';

import { library } from '../../actions';

const mapDispatchToProps = {
  createLibrary: library.create.dispatch,
  editLibrary: library.update.dispatch,
};

export default connect(
  null,
  mapDispatchToProps
)(Library);
