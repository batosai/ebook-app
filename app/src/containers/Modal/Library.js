import { connect } from 'react-redux';
import Library from '../../components/Modal/Library';

import { libraryAddRequestAction, libraryEditRequestAction } from '../../actions';

const mapDispatchToProps = dispatch => ({
  createLibrary: library => dispatch(libraryAddRequestAction(library)),
  editLibrary: library => dispatch(libraryEditRequestAction(library)),
});

export default connect(
  null,
  mapDispatchToProps
)(Library);
