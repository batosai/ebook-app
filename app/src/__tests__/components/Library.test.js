import React from 'react';
import Library from '../../components/Library';
import { shallow, mount, render } from 'enzyme';

jest.dontMock('../../components/Library');

// TODO enzyme pour utiliser un context et Ã©viter le bug de react-test-renderer
// => prepareStyles
// import getMuiTheme from 'material-ui/styles/getMuiTheme';
//http://redux.js.org/docs/recipes/WritingTests.html
// https://github.com/callemall/material-ui/issues/4664
//https://github.com/callemall/material-ui/issues/5330

describe('Components', () => {
  it('Library', () => {
    // const expectedAction = createAction(types.LIBRARIES_REQUEST, () => null, {emit:true});
    //
    // expect(actions.librariesRequest()).toEqual(expectedAction());
    // expect(actions.librariesRequest()).toMatchSnapshot();
  });


  // const style = {
  //   root: {},
  //   gridList: {}
  // };
  // const component = renderer.create(
  //   <Library style={style} tiles={[]} />
  // );
  // let tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
});
