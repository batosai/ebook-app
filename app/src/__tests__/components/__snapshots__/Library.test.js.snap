exports[`Components Library 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    style={
        Object {
            "display": "flex",
            "flexWrap": "wrap",
            "justifyContent": "space-around",
          }
    }>
    <GridList
        cellHeight={180}
        cols={2}
        padding={4}
        style={
            Object {
                "height": "100%",
                "overflowY": "auto",
                "width": "100%",
              }
        }>
        <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/collection/1">
            <GridTile
                actionPosition="right"
                cols={1}
                containerElement="div"
                rows={1}
                subtitle={
                    <span>
                        by 
                        <b>
                            jeremy
                        </b>
                    </span>
                }
                title="run test"
                titleBackground="rgba(0, 0, 0, 0.4)"
                titlePosition="bottom">
                <img
                    alt=""
                    src="" />
            </GridTile>
        </Link>
        <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/collection/2">
            <GridTile
                actionPosition="right"
                cols={1}
                containerElement="div"
                rows={1}
                subtitle={
                    <span>
                        by 
                        <b>
                            jeremy
                        </b>
                    </span>
                }
                title="run test 2"
                titleBackground="rgba(0, 0, 0, 0.4)"
                titlePosition="bottom">
                <img
                    alt=""
                    src="" />
            </GridTile>
        </Link>
    </GridList>
</div>,
  "nodes": Array [
    <div
      style={
            Object {
                  "display": "flex",
                  "flexWrap": "wrap",
                  "justifyContent": "space-around",
                }
      }>
      <GridList
            cellHeight={180}
            cols={2}
            padding={4}
            style={
                  Object {
                        "height": "100%",
                        "overflowY": "auto",
                        "width": "100%",
                      }
            }>
            <Link
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/collection/1">
                  <GridTile
                        actionPosition="right"
                        cols={1}
                        containerElement="div"
                        rows={1}
                        subtitle={
                              <span>
                                    by 
                                    <b>
                                          jeremy
                                    </b>
                              </span>
                        }
                        title="run test"
                        titleBackground="rgba(0, 0, 0, 0.4)"
                        titlePosition="bottom">
                        <img
                              alt=""
                              src="" />
                  </GridTile>
            </Link>
            <Link
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/collection/2">
                  <GridTile
                        actionPosition="right"
                        cols={1}
                        containerElement="div"
                        rows={1}
                        subtitle={
                              <span>
                                    by 
                                    <b>
                                          jeremy
                                    </b>
                              </span>
                        }
                        title="run test 2"
                        titleBackground="rgba(0, 0, 0, 0.4)"
                        titlePosition="bottom">
                        <img
                              alt=""
                              src="" />
                  </GridTile>
            </Link>
      </GridList>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Library
        style={
                Object {
                        "gridList": Object {
                          "height": "100%",
                          "overflowY": "auto",
                          "width": "100%",
                        },
                        "root": Object {
                          "display": "flex",
                          "flexWrap": "wrap",
                          "justifyContent": "space-around",
                        },
                      }
        }
        tiles={
                Array [
                        Object {
                          "author": "jeremy",
                          "id": 1,
                          "img": "",
                          "library_id": 1,
                          "title": "run test",
                        },
                        Object {
                          "author": "jeremy",
                          "id": 2,
                          "img": "",
                          "library_id": 2,
                          "title": "run test 2",
                        },
                      ]
        } />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Library {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "style": Object {
            "gridList": Object {
              "height": "100%",
              "overflowY": "auto",
              "width": "100%",
            },
            "root": Object {
              "display": "flex",
              "flexWrap": "wrap",
              "justifyContent": "space-around",
            },
          },
          "tiles": Array [
            Object {
              "author": "jeremy",
              "id": 1,
              "img": "",
              "library_id": 1,
              "title": "run test",
            },
            Object {
              "author": "jeremy",
              "id": 2,
              "img": "",
              "library_id": 2,
              "title": "run test 2",
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          style={
                    Object {
                              "display": "flex",
                              "flexWrap": "wrap",
                              "justifyContent": "space-around",
                            }
          }>
          <GridList
                    cellHeight={180}
                    cols={2}
                    padding={4}
                    style={
                              Object {
                                        "height": "100%",
                                        "overflowY": "auto",
                                        "width": "100%",
                                      }
                    }>
                    <Link
                              onlyActiveOnIndex={false}
                              style={Object {}}
                              to="/collection/1">
                              <GridTile
                                        actionPosition="right"
                                        cols={1}
                                        containerElement="div"
                                        rows={1}
                                        subtitle={
                                                  <span>
                                                            by 
                                                            <b>
                                                                      jeremy
                                                            </b>
                                                  </span>
                                        }
                                        title="run test"
                                        titleBackground="rgba(0, 0, 0, 0.4)"
                                        titlePosition="bottom">
                                        <img
                                                  alt=""
                                                  src="" />
                              </GridTile>
                    </Link>
                    <Link
                              onlyActiveOnIndex={false}
                              style={Object {}}
                              to="/collection/2">
                              <GridTile
                                        actionPosition="right"
                                        cols={1}
                                        containerElement="div"
                                        rows={1}
                                        subtitle={
                                                  <span>
                                                            by 
                                                            <b>
                                                                      jeremy
                                                            </b>
                                                  </span>
                                        }
                                        title="run test 2"
                                        titleBackground="rgba(0, 0, 0, 0.4)"
                                        titlePosition="bottom">
                                        <img
                                                  alt=""
                                                  src="" />
                              </GridTile>
                    </Link>
          </GridList>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          style={
                    Object {
                              "display": "flex",
                              "flexWrap": "wrap",
                              "justifyContent": "space-around",
                            }
          }>
          <GridList
                    cellHeight={180}
                    cols={2}
                    padding={4}
                    style={
                              Object {
                                        "height": "100%",
                                        "overflowY": "auto",
                                        "width": "100%",
                                      }
                    }>
                    <Link
                              onlyActiveOnIndex={false}
                              style={Object {}}
                              to="/collection/1">
                              <GridTile
                                        actionPosition="right"
                                        cols={1}
                                        containerElement="div"
                                        rows={1}
                                        subtitle={
                                                  <span>
                                                            by 
                                                            <b>
                                                                      jeremy
                                                            </b>
                                                  </span>
                                        }
                                        title="run test"
                                        titleBackground="rgba(0, 0, 0, 0.4)"
                                        titlePosition="bottom">
                                        <img
                                                  alt=""
                                                  src="" />
                              </GridTile>
                    </Link>
                    <Link
                              onlyActiveOnIndex={false}
                              style={Object {}}
                              to="/collection/2">
                              <GridTile
                                        actionPosition="right"
                                        cols={1}
                                        containerElement="div"
                                        rows={1}
                                        subtitle={
                                                  <span>
                                                            by 
                                                            <b>
                                                                      jeremy
                                                            </b>
                                                  </span>
                                        }
                                        title="run test 2"
                                        titleBackground="rgba(0, 0, 0, 0.4)"
                                        titlePosition="bottom">
                                        <img
                                                  alt=""
                                                  src="" />
                              </GridTile>
                    </Link>
          </GridList>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Library
    style={
        Object {
            "gridList": Object {
              "height": "100%",
              "overflowY": "auto",
              "width": "100%",
            },
            "root": Object {
              "display": "flex",
              "flexWrap": "wrap",
              "justifyContent": "space-around",
            },
          }
    }
    tiles={
        Array [
            Object {
              "author": "jeremy",
              "id": 1,
              "img": "",
              "library_id": 1,
              "title": "run test",
            },
            Object {
              "author": "jeremy",
              "id": 2,
              "img": "",
              "library_id": 2,
              "title": "run test 2",
            },
          ]
    } />,
}
`;
